File: jmix-templates/content/component/custom-component/loader.java
package ${packageName};

import io.jmix.flowui.xml.layout.loader.AbstractComponentLoader;

public class ${loaderClassName} extends AbstractComponentLoader<${componentClassName}> {

    @Override
    protected ${componentClassName} createComponent() {
        return factory.create(${componentClassName}.class);
    }

    @Override
    public void loadComponent() {
        // TODO load your component's XML attributes, e.g.:
        // loadString(element, "value", resultComponent::setValue);
        componentLoader().loadClassNames(resultComponent, element);
    }
}

File: jmix-templates/content/component/custom-component/loader.kt
package ${packageName}

import io.jmix.flowui.xml.layout.loader.AbstractComponentLoader

class ${loaderClassName} : AbstractComponentLoader<${componentClassName}>() {

    override fun createComponent(): ${componentClassName} = factory.create(${componentClassName}::class.java)

    override fun loadComponent() {
        // TODO load your component's XML attributes, e.g.:
        // loadString(element, "value") { resultComponent.value = it }
        componentLoader().loadClassNames(resultComponent, element)
    }
}
